src/main.rs

```diff
@@ -1,3 +1,132 @@
+#[macro_use]
+extern crate clap;
+
+mod options;
+use crate::options::CLIOptions;
+use clap::App;
+use std::{fs, fs::File, io, path::Path};
+
 fn main() {
-    println!("Hello, world!!!");
+    let yaml = load_yaml!("../cli_def/en_us.yml");
+    let matches = App::from_yaml(yaml).get_matches();
+
+    let raw_options = CLIOptions::from_clap_matches(&matches);
+
+    let options = raw_options.process_options();
+
+    for filename in options.files {
+        let file = File::open(filename).unwrap();
+        unzip_file(
+            file,
+            &std::path::PathBuf::from(&raw_options.destination_folder),
+        );
+    }
+}
+
+fn unzip_file(file: File, directory: &Path) {
+    let mut archive = zip::ZipArchive::new(file).unwrap();
+
+    let files_indexes = 0..archive.len();
+
+    files_indexes
+        .map(|file_index| unzip_archive(file_index, &mut archive, directory))
+        .for_each(drop);
+}
+
+fn unzip_archive(file_index: usize, archive: &mut zip::ZipArchive<File>, directory: &Path) {
+    let mut file = archive.by_index(file_index).unwrap();
+    let outpath = file.sanitized_name();
+
+    {
```

Comment:
I am not sure also why would you need another scope for this?

Reply:
All that functions is a copy paste from the example :P

I will clean it up a bit.